test-framework/junit5-internal/src/main/java/io/quarkus/test/ExportUtil.java

```diff
@@ -35,4 +46,41 @@ static void exportToQuarkusDeploymentPath(JavaArchive archive) throws IOExceptio
         File exportFile = new File(exportDir, archive.getName());
         archive.as(ZipExporter.class).exportTo(exportFile);
     }
+
+    static void mergeCustomApplicationProperties(JavaArchive archive, Properties customApplicationProperties)
+            throws IOException {
+        Node applicationProperties = archive.get(APPLICATION_PROPERTIES);
+        if (applicationProperties != null) {
+            // Merge the existing "application.properties" asset and overriden config properties
+            // Overriden properties take precedence
+            Properties mergedProperties = new Properties();
+            Asset asset = applicationProperties.getAsset();
+            if (asset instanceof StringAsset strAsset) {
+                mergedProperties.load(new StringReader(strAsset.getSource()));
+            } else {
+                try (InputStream in = asset.openStream()) {
+                    mergedProperties.load(in);
+                }
+            }
+            customApplicationProperties.forEach(mergedProperties::put);
+
+            if (Boolean.parseBoolean(System.getProperty("quarkus.test.log-merged-properties"))) {
```

Comment:
This can also be simplified to: 

```suggestion
            if (Boolean.getBoolean("quarkus.test.log-merged-properties")) {
```

Reply:
Ok, will use it the next time ;-). Although I personally don't think `getBoolean()` is a good method name. I mean `parseSystemProperty()` or something along these lines would be more clear :shrug:.

Reply:
We need to blame Sun for introducing this method in Java 1.0 😃

Reply:
I'm with @mkouba on this - `Boolean.getBoolean(...)` is super confusing in that it parses a system property.